src/interactive-window/interactiveWindowProvider.ts

```diff
@@ -132,6 +156,15 @@ export class InteractiveWindowProvider
         if (!result) {
             // No match. Create a new item.
             result = await this.create(resource, mode, connection);
+            // We already check preferred kernel before creating the interactive window
+            // no need to update it anymore.
+            await result.start(undefined);
+        } else {
+            const preferredController = connection
```

Comment:
I think we should cache the controller id as well.
Assume user starts IW, and then changes the kernel, now they reload VS Code, the IW will point to the wrong kernel.

Reply:
Good catch

> changes the kernel, now they reload VS Code

To double confirm, there is no "cell execution" between these two steps, right?

Reply:
Cell execution happens after the window is created, so yeah there shouldn't be any cell execution between these two steps.